Create chat views between users

rupesh-ps opened the pull request:
## PR Title: Create chat views between users 

### Description

This PR includes the below:
- **Create views for chat**: Created `chat_detail` and `inbox` views with their templates.
-  **Edit/Delete feature**: User can edit/delete their messages with a time constraint.
- **User Testing**: Verified the functionality of views with real users.
- **Quick Message Button**: Added a "new message" button under user profiles to start new chat, easily.

rupesh-ps committed:
Create views for Chat

rupesh-ps committed:
Design templates for chat and inbox views

rupesh-ps committed:
Test views and their functionality w.r.t users

rupesh-ps committed:
Modify templates chat

rupesh-ps committed:
Added new message quick button under user profile

rupesh-ps committed:
Add instant new message for better UX

shantanutomarr commented on adBridge/ad_chat/views.py:
```diff
@@ -1,3 +1,114 @@
+    return redirect('ad_chat:chat_detail', chat_id=chat.id)
+
+@login_required
+def chat_detail(request, chat_id):
```
Write this as class-based view with small methods. Extract code into smaller chunks with meaningful names.

rupesh-ps replied:
Refactored the code.

rupesh-ps committed:
Refactor views to use class-based structure

- Converted views to class-based with `ChatMixin`,
 `MessageService` & `MessagePermissions` static
 methodsfor cleaner logic and reusable methods
- Improved message editing and deletion functionality.
- Removed message cloning and interval based scrolling.

shantanutomarr commented on adBridge/ad_chat/views.py:
```diff
@@ -1,3 +1,153 @@
+    def get_object_by_id(self, model, object_id):
+        return get_object_or_404(model, id=object_id)
+
+class MessageService:
```
Since this is used by only a single view, you can move it inside the vide itself. No need for separate class.

shantanutomarr commented on adBridge/ad_chat/views.py:
```diff
@@ -1,3 +1,153 @@
+            return True
+        return False
+
+class MessagePermissions:
```
Same as above

shantanutomarr commented on adBridge/ad_chat/views.py:
```diff
@@ -1,3 +1,153 @@
+        message_text = request.POST.get('message')
+        message_id = request.POST.get('message_id')
+
+        if message_text:
+            if message_id:
+                message = get_object_or_404(ChatMessage, id=message_id)
+                if MessageService.update_message(message, message_text, request.user):
+                    messages.success(request, 'Message updated successfully.')
+                else:
+                    messages.error(request, 'You can only edit your messages within 2 minutes of sending.')
+            else:
+                return self.send_new_message(request, message_text, chat)
```
Extract to separate method with meaningful name.

shantanutomarr commented on adBridge/ad_chat/views.py:
```diff
@@ -1,3 +1,153 @@
+            else:
+                return self.send_new_message(request, message_text, chat)
+
+        elif 'delete_message' in request.POST:
+            message_id = request.POST.get('delete_message')
+            message = get_object_or_404(ChatMessage, id=message_id)
+            if MessageService.delete_message(message, request.user):
+                messages.success(request, 'Message deleted successfully.')
+            else:
+                messages.error(request, 'You can only delete your own messages.')
```
Extract to separate method with meaningful name.

rupesh-ps committed:
Refactor the code

- Extract message update, delete, and send logic
into separate helper methods
- Add handler methods for message-related actions
- Reuse `can_edit` and `can_delete` methods to avoid
 duplication
